### ISSUE TITLE
Need detailed connection pics

### ISSUE THREAD

**Issue by senolsun** [HeatseekerXXX/Dyson_UART_parser#1]
Hello. First of all, thank you very much for sharing such an amazing project with all of us. I was quite surprised when I googled dyson esp32 fan and found your project which is exactly what I was aiming for.

I am a newbie in electronics and would like to see a few more photos of the electronic board connections if you have them available. Especially the part where you suggest us to use a 100R resistor, the photo is not clear as you drew on it so I am not sure where the resistor goes to. Thanks!

**Comment by zcshiner** [comment]
Agreed.  I'd love to see a bit more documentation where your UART connections are made.  I am trying to replicate this with model TP01 / AM07 and I'm not having luck.

**Comment by cynster** [comment]
It would be great to have a more detailed description (including photo's), since I'm pretty new to this as well. Might give it a try on my TP7A.

**Comment by aarongorka** [comment]
I don't have mine hooked up yet but I've taken it apart and I've found what I think is the UART port.

  1. First you need to take out the screws inside where the filter goes. Don't bother with the screws on the very bottom.
  ![Image](https://github.com/user-attachments/assets/c0e11c32-f5d5-489d-986f-7ab8fd246d90)
  1. The screen slides up
  ![Image](https://github.com/user-attachments/assets/9cf50dfe-3590-4312-80d4-95c4c02ac3f8)
  1. Slide the processing unit out, detach the cable
  ![Image](https://github.com/user-attachments/assets/4a06cac9-a265-43aa-b85b-dbb83ce6cbc4)
  1. What I believe is the UART port is on the opposite side of the screen, located below what looks like a reset button. There aren't any other unconnected ports, and it has 4 pins as per the README. It's small, much smaller than a standard JST connector. I'm thinking maybe 1mm or 1.25mm pitch?
  ![Image](https://github.com/user-attachments/assets/56ff2ec9-a509-49b8-8d28-5f7d6c7445d6)

I'm going to see if I can buy a cable and will report back.

**Comment by aarongorka** [comment]
I can confirm that it's a 1mm pitch JST cable port (1.25mm does not fit). As I put it back together I also realised there's actually a hole right under the display that lets you access the UART port (but if you want to solder the IR trace, you'll need to disassemble it anyway).

![Image](https://github.com/user-attachments/assets/8b47cfab-65ef-46e3-a26c-e9ec0fb557c3)

 I can also confirm the diagram with the pinout of the UART is correct (in my photo red is 3.3v, green is ground, yellow is TX and the disconnected blue is RX). The code also seems to check out. While I didn't use it directly, I did use it as inspiration for an esphome component which I have working.

**Comment by zcshiner** [comment]
Just to confirm, you are plugging into the connector labeled J12 on the board?

**Comment by aarongorka** [comment]
> Just to confirm, you are plugging into the connector labeled J12 on the board?

Yes.

**Comment by b1scuitdev** [comment]
Can the device be turned on by using only the UART port? What is the purpose of connecting ESP32 to the IR line?

**Comment by aarongorka** [comment]
> Can the device be turned on by using only the UART port? What is the purpose of connecting ESP32 to the IR line?

I'd imagine the firmware does have that capability, but it hasn't been reverse engineered, so - no. IR is the only way to control it.

Also, while I'm here - if anyone's interested, I've published my ESPHome external component here https://github.com/aarongorka/esphome-components/tree/main#dyson-tp07-component
